### PR TITLE
Implementation of lofreq somatic and callparallel to sarek as variant callers 

### PR DESCRIPTION
**Pull request by AitorPeseta**
#1588 

- [X] This comment contains a description of changes (with reason) .
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [X] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nf-test test tests/ --verbose --profile +docker`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [X] Usage Documentation in `docs/usage.md` is updated.
- [X] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

We wanted to ask you for a **code review** in this PR. We are updating Sarek with new modules: `lofreq somatic` (paired sample variant calling) and `lofreq callparallel` (tumor-only variant calling). Therefore, there will be two new subworkflows: one for somatic and the other one for tumor-only. 

**In order to these modules fitting in Sarek, we have made some changes in them in this fork**. Consequently,  they are not exactly the same modules from nf-core modules repo (they will be updated when we finish this). For this reason, we will be working in parallel to update these modules in nf-core/modules.

To sum up, we wanted to ask for a code review, specifically in these new `lofreq` subworkflows.

### CODE REVIEW

**Comment by nevinwu** [comment]
We have implemented `lofreq call-parallel` as a *variant caller* for `sarek`. We have included `bedtools sort` and `bedtools merge` in a `intervals.bed` processing step to ensure the bed file is properly formed. If it is not, lofreq raises an error. Please let us know if we should keep or remove this bed-processing step.

Unfortunately, we weren't able to implement `lofreq somatic` due to the way this software works. It generates several "intermediate" files and, if one of these *doesn't have somatic variants* (which is indeed pretty possible as we've checked with targetseq panel data) `lofreq` raises and error. Consequently, nextflow stops its execution.

**Comment by nevinwu** [comment]
@nf-core-bot fix linting

**Comment by AitorPeseta** [comment]
@nf-core-bot fix linting

**Comment by nevinwu** [comment]
@nf-core-bot fix linting pretty please

**Comment by AitorPeseta** [comment]
Fran and I have decided to close this PR and make the same changes in the Dev branch instead of the master, as it seems that Prettier has changed many files, and that was not our intention.
